### PR TITLE
Update terraform config and create github action workflow for this

### PR DESCRIPTION
**Pull request by Fastjur**

### CODE REVIEW

File: terraform/main.tf [hunk 1]
```diff
@@ -1,12 +1,63 @@
...
-    google = {
-      source = "hashicorp/google"
-      version = "4.20.0"
-    }
+provider "google" {
+  project = var.project_id
+  region  = "europe-west1"
+}
+
+module "gke_auth" {
```
Comment (Fastjur) [inline terraform/main.tf:6]: My IntelliJ is complaining about the fact that these modules do not exist. I think the proper usage of terraform modules is linked to a directory? Do we care?

Comment (pietertolsma) [reply terraform/main.tf:6]: It's not complaining in VS Code, and since it is working I dont really mind tbh

**Review by pietertolsma — APPROVED** [review]

**Review by pietertolsma — CHANGES_REQUESTED** [review]
This PR has the wrong version of inference. Check out the new PR

**Comment by Fastjur** [comment]
Superseded by https://github.com/Fastjur/nidhogg/pull/3

**Pull request closed without merging**